sycl-jit/jit-compiler/include/Options.h

```diff
@@ -45,6 +45,8 @@ class OptionStorage {
   }
 
   OptionStorage &operator=(OptionStorage &&Other) {
+    if (this == &Other)
+      return *this;
     Storage = Other.Storage;
```

Comment:
If `Storage` was non-null in the assigned-to object, that memory will leak. Should we rather fix that 1) instead of, or 2) in addition, to the self-assignment protection? The difference, AFAICT, is whether `x = std::move(x)` leaves `x` untouched or just in unitialised state.

Reply:
Thank you for spotting this. I'm re-reading the complaint from the static analyser and wondering if we actually need to do anything with it. I was too quick to jump to a fix I guess, compiler should warn against such a move (under `-Wself-move` for both clang and gcc). Tempted to remove that check and mark it as false-positive in the tool.

Reply:
Reverting and marking as false positiv SGTM! I'll make a note to look into the potential memory leak.

Reply:
Done